packages/protocol/contracts/stability/SortedOracles.sol

```diff
@@ -118,6 +118,21 @@ contract SortedOracles is ISortedOracles, Ownable, Initializable {
     }
   }
 
+  /**
+   * @notice Check if last report is expired.
+   * @param token The address of the token for which the Celo Gold exchange rate is being reported.
+   */
+  function isReportActive(address token) external view returns (bool) {
+    require(token != address(0));
+    address oldest = timestamps[token].getTail();
+    uint256 timestamp = timestamps[token].getValue(oldest);
+    // solhint-disable-next-line not-rely-on-time
+    if (now.sub(timestamp) >= reportExpirySeconds) {
+      return false;
+    }
+    return true;
```

Comment:
Since this is the same as the check that's used in `removeExpiredReports`, could this reasonably be used inside that function?

Are there any complexities with `external` vs `public` that make code duplication preferable?

Reply:
+1, `removeExpiredReports` can remove the oldest reports until `isOldestReportExpired` returns false